### PR TITLE
additional cleaning up of data upon logging out and accept nil respon…

### PR DESCRIPTION
**Pull request by thuyBox**
…se data for a successful bg upload
and support reconnection to background sessions created from extensions

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @thuyBox has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/Operations/BOXAPIMultipartToJSONOperation.m [hunk 1]
```diff
@@ -132,6 +132,16 @@ - (void)sessionTask:(NSURLSessionTask *)sessionTask processIntermediateData:(NSD
     }
 }
 
+- (void)sessionTask:(NSURLSessionTask *)sessionTask didFinishWithResponse:(NSURLResponse *)response responseData:(NSData *)responseData error:(NSError *)error
+{
+    @synchronized (self) {
+        if ([self shouldRunInBackground] == YES) {
+            [[NSFileManager defaultManager] removeItemAtPath:self.uploadMultipartCopyFilePath error:nil];
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Operations/BOXAPIMultipartToJSONOperation.m:139]: Do we not need the copy on retries for failures/etc?

**Review by boxtcarpel — APPROVED** [review]

File: BoxContentSDK/BoxContentSDK/Operations/BOXAPIMultipartToJSONOperation.m (hunk 1 above)
Comment (boxtcarpel) [reply BoxContentSDK/BoxContentSDK/Operations/BOXAPIMultipartToJSONOperation.m:139]: We also discussed offline about centralizing cleanup (finish?)

**Comment by boxtcarpel** [comment]
We also spoke offline about forcing cleanup, making sure more that everything on disk goes away at logout.

File: BoxContentSDK/BoxContentSDK/Operations/BOXAPIMultipartToJSONOperation.m (hunk 1 above)
Comment (thuyBox) [reply BoxContentSDK/BoxContentSDK/Operations/BOXAPIMultipartToJSONOperation.m:139]: we won't be able to retry upload using the same operation once it's finished. User can create a new request and the temp file will be re-created.

**Comment by thuyBox** [comment]
will follow up with a PR to force clean up of userId cache upon logout even if not empty, and handle background tasks without a userId gracefully.

**Pull request merged**